### PR TITLE
Op creation time

### PR DESCRIPTION
**Pull request by towards-a-new-leftypol**
Rate limit new threads being created to 40 seconds between each. Regardless of IP.

Unfortunately this isn't quite the same as rate-limiting the number of new threads per hour created... but it's the next best thing.

### CODE REVIEW

File: inc/instance-config.php [hunk 1]
```diff
@@ -82,9 +96,12 @@
 $config['cookies']['mod'] = 'mod';
 $config['cookies']['salt'] = 'MGYwNjhlNjU5Y2QxNWU3YjQ3MzQ1Yj';
 
-$config['flood_time'] = 30;
-$config['flood_time_ip'] = 60;
-$config['flood_time_same'] = 60;
+
+$config['flood_cache'] = 60 * 15; // 15 minutes
+$config['flood_time_any'] = 40; // in seconds
+$config['flood_time'] = 0;
```
Comment (towards-a-new-leftypol) [inline inc/instance-config.php]: these need to be changed back to 30 or 60, the only new value added here is flood_time_any

File: inc/instance-config.php [hunk 2]
```diff
@@ -24,6 +24,20 @@
   array('meta')
 );
 
+// /*
```
Comment (towards-a-new-leftypol) [inline inc/instance-config.php]: this commented block is garbage

**Comment by towards-a-new-leftypol** [comment]
These changes are in production. FYI

File: inc/filters.php [hunk 3]
```diff
@@ -9,245 +9,271 @@
...
+                    $flood_check_matched[] = $flood_post;
+                }
+                
+                // is there any reason for this assignment?
+                $this->flood_check = $flood_check_matched;
+                
+                return !empty($this->flood_check);
+            case 'flood-time-any':
+                foreach ($this->flood_check as $flood_post) {
+                    if (time() - $flood_post['time'] <= $match) {
```
Comment (nonmakina) [inline inc/filters.php:78]: Should this line be:
`if (time() - $flood_post['time'] <= $match && !$flood_post[isreply]) {`

**Review by nonmakina — APPROVED** [review]

**Pull request merged**